src/gp_engine.cpp

```diff
@@ -172,7 +173,7 @@ DoublePair GPEngine::LogLikelihoodAndDerivative(
   // The phylogenetic component of the likelihood is weighted with the number of times
   // we see the site patterns.
   const double log_likelihood =
-      log(q_(op.gpcsp_)) + per_pattern_log_likelihoods_.dot(site_pattern_weights_);
+      (per_pattern_log_likelihoods_.array() * site_pattern_weights_.array()).sum();
```

Comment:
I'd like to know more about this change-- why we are dropping the q component. 

Also, isn't the new code equivalent to a .dot()?

Reply:
`q` is dropped because it is incorporated into `per_pattern_log_likelihoods_` in `PreparePerPatternLogLikelihoodsForGPCSP`:

https://github.com/phylovi/libsbn/blob/4eabe4405edcc56e82c94e0a701cd8baa30fd863/src/gp_engine.hpp#L170-L179

Dropping `q` here is one of the fallouts from realizing that we aren't doing exact marginal but a composite marginal. 

Yes, it is updated to use `.dot()`.